plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/service/RulePersistenceService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ResourceAlreadyExistsException;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.plugin.wlm.rule.action.CreateRuleResponse;
+import org.opensearch.wlm.Rule;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This class defines the functions for Rule persistence
+ * @opensearch.experimental
+ */
+public class RulePersistenceService {
+    public static final String RULE_INDEX = ".rule";
+    private final Client client;
+    private final ClusterService clusterService;
+    private static final Logger logger = LogManager.getLogger(RulePersistenceService.class);
+
+    /**
+     * Constructor for RulePersistenceService
+     * @param client {@link Client} - The client to be used by RulePersistenceService
+     */
+    @Inject
+    public RulePersistenceService(final ClusterService clusterService, final Client client) {
+        this.clusterService = clusterService;
+        this.client = client;
+    }
+
+    /**
+     * This method is the entry point for create rule logic in persistence service.
+     * @param rule - the rule that we're persisting
+     * @param listener - ActionListener for CreateRuleResponse
+     */
+    public void createRule(Rule rule, ActionListener<CreateRuleResponse> listener) {
+        createRuleIndexIfAbsent(new ActionListener<>() {
+            @Override
+            public void onResponse(Boolean indexCreated) {
+                persistRule(rule, listener);
```

Comment:
We are not using the method param here. I think either we should remove this or make use of it to justify it in the method signature